### PR TITLE
Topology2: Add sof-lnl-rt712-l3-rt1712-l1 topology

### PR DESCRIPTION
**Pull request by aiChaoSONG**
This patch adds the topology for LNL RVP with rt712 codec.

### CODE REVIEW

**Comment by aiChaoSONG** [comment]
Linux PR: https://github.com/thesofproject/linux/pull/4760
Headset playback & capture, Speaker playback and DMIC feature are verified manually.

**Review by bardliao — APPROVED** [review]

File: tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake [hunk 1]
```diff
@@ -104,4 +104,8 @@ DEEPBUFFER_FW_DMA_MS=100,DEEPBUFFER_D0I3_COMPATIBLE=true"
 
 "cavs-sdw\;sof-lnl-rt711-l0-rt1316-l23-rt714-l1\;PLATFORM=lnl,NUM_SDW_AMP_LINKS=2,SDW_DMIC=1,\
 NUM_HDMIS=0,SDW_SPK_STREAM=SDW2-Playback,SDW_SPK_IN_STREAM=SDW2-Capture,SDW_DMIC_STREAM=SDW1-Capture"
+
+"cavs-sdw\;sof-lnl-rt712-l3-rt1712-l1\;PLATFORM=lnl,SDW_DMIC=1,NUM_HDMIS=0,NUM_SDW_AMP_LINKS=1,\
+SDW_AMP_FEEDBACK=false,SDW_SPK_STREAM=Playback-SmartAmp,SDW_DMIC_STREAM=Capture-SmartMic,\
```
Comment (plbossart) [inline tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake:109]: paranoia check: are you sure there's no amp feedback? all recent amps have such a stream.

Comment (aiChaoSONG) [reply tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake:109]: The codec spec says amp feedback is supported, but there is no feedback DAI declared in codec driver.

Comment (plbossart) [reply tools/topology/topology2/sof-ace-tplg/tplg-targets.cmake:109]: ok, thanks @aiChaoSONG 

We'll probably need an update at some point but there's plenty of time to deal with this later.

**Review by plbossart — APPROVED** [review]

**Comment by aiChaoSONG** [comment]
Let's on hold to merge this patch, the support for LNL RVP + RT722 codec is developed against a BIOS from Windows BKC from WW48, however, when I upgrade BIOS to WW50 version from Windows BKC, the link configuration is changed, this is way frustrating. I will try more BIOS to check the link configuration and align with Keqiao what BIOS we will use in CI in the future, and then come back to update the patch.

https://github.com/thesofproject/linux/pull/4760 may need a fixup.

**Comment by aiChaoSONG** [comment]
Linux fixup: https://github.com/thesofproject/linux/pull/4762, and this patch is updated accordingly.

**Pull request merged**